### ISSUE TITLE
Failed whenever i upload locally

### ISSUE THREAD

**Issue by ea2399** [stemrollerapp/stemroller#74]
Im having the issue that anytime i upload anything local to Stemroller it immediately fails to convert. The naming of the file does not matter either. But when i try to search the song, it works.

Stemroller 2.0.7
Windows 11 Pro

**Comment by lunndal** [comment]
I'm seeing the same issue, but for me it also fails on downloads.

**Comment by ea2399** [comment]
ya im noticing now that its also failing on downloads. it wasnt doing this previously. I couldnt find any log files either, also when i try to run it using CMD

**Comment by lunndal** [comment]
It seems local files worked for me when i tried MP3 instead of AIFF as _input_ format. I have edited my comment accordingly :)

**Comment by iffyloop** [comment]
Sorry to hear about these issues. I doubt this will fix it, but try upgrading to `2.1.0` and see if things have changed.
Thanks @lunndal for your detailed reports - it is strange to me that local files are working on Win11 but not 10. Thankfully, the download error you were experiencing before should be fixed on both platforms.

**Comment by lunndal** [comment]
Thanks for getting back to us on this, @iffyloop . It's much appreciated. Never say sorry, mate - we're just super happy you're doing this. This is a piece of awesome software, and we're privileged to recieve the fruits of your labours 🥳

I've tested v2.1.1 on Win10 Ent 22H2, but now it fails on both local then download. Here is the console output:

```
Microsoft Windows [Version 10.0.19045.4651]
(c) Microsoft Corporation. All rights reserved.

C:\WINDOWS\system32>c:\Users\USERNAME\Downloads\stemroller-2.1.1-win-cuda\StemRoller.exe

C:\WINDOWS\system32>
Successfully blocked power-save using policy: "prevent-app-suspension"
BEGIN downloading/processing video "88cc588dcb2b9b0d" - "Free Bird - Lynyrd Skynyrd"
Splitting video "88cc588dcb2b9b0d"; 4 jobs using model "htdemucs"...
Running with "-d cpu" to force CPU instead of CUDA
Trace: Error: ENOENT: no such file or directory, scandir 'C:\Users\USERNAME\AppData\Local\Temp\StemRoller-XXXXXXOeg6Es\separated\htdemucs'
    at async readdir (node:internal/fs/promises:948:18)
    at async findDemucsOutputDir (c:\Users\USERNAME\Downloads\stemroller-2.1.1-win-cuda\resources\app.asar\main-src\processQueue.cjs:188:19)
    at async _processVideo (c:\Users\USERNAME\Downloads\stemroller-2.1.1-win-cuda\resources\app.asar\main-src\processQueue.cjs:324:26)
    at async processVideo (c:\Users\USERNAME\Downloads\stemroller-2.1.1-win-cuda\resources\app.asar\main-src\processQueue.cjs:440:5) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'scandir',
  path: 'C:\\Users\\USERNAME\\AppData\\Local\\Temp\\StemRoller-XXXXXXOeg6Es\\separated\\htdemucs'
}
    at processVideo (c:\Users\USERNAME\Downloads\stemroller-2.1.1-win-cuda\resources\app.asar\main-src\processQueue.cjs:442:13)
Successfully unblocked power-save
Successfully blocked power-save using policy: "prevent-app-suspension"
BEGIN downloading/processing video "0LwcvjNJTuM" - "Lynyrd Skynyrd - Free Bird (Official Audio)"
Downloading YouTube video "0LwcvjNJTuM"; storing in "C:\Users\USERNAME\AppData\Local\Temp\StemRoller-XXXXXXMRd2SV\yt-audio"
Splitting video "0LwcvjNJTuM"; 4 jobs using model "htdemucs"...
Running with "-d cpu" to force CPU instead of CUDA
Trace: Error: ENOENT: no such file or directory, scandir 'C:\Users\USERNAME\AppData\Local\Temp\StemRoller-XXXXXXMRd2SV\separated\htdemucs'
    at async readdir (node:internal/fs/promises:948:18)
    at async findDemucsOutputDir (c:\Users\USERNAME\Downloads\stemroller-2.1.1-win-cuda\resources\app.asar\main-src\processQueue.cjs:188:19)
    at async _processVideo (c:\Users\USERNAME\Downloads\stemroller-2.1.1-win-cuda\resources\app.asar\main-src\processQueue.cjs:324:26)
    at async processVideo (c:\Users\USERNAME\Downloads\stemroller-2.1.1-win-cuda\resources\app.asar\main-src\processQueue.cjs:440:5) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'scandir',
  path: 'C:\\Users\\USERNAME\\AppData\\Local\\Temp\\StemRoller-XXXXXXMRd2SV\\separated\\htdemucs'
}
    at processVideo (c:\Users\USERNAME\Downloads\stemroller-2.1.1-win-cuda\resources\app.asar\main-src\processQueue.cjs:442:13)
Successfully unblocked power-save
```

Thanks, mate!

**Comment by lunndal** [comment]
Hi @iffyloop! New update from me.

I did a clean install of Windows 11 Pro (23H2) and ran v2.1.1 in an admin shell. Fails on both local files and download. 

Here are the console logs:

```
#
# Input mp3 as local file - Ver 2.1.1 on Win11 Pro 23H2, clean install, shell as administrator
#

C:\share>.\stemroller-2.1.1-win-cuda\StemRoller.exe

C:\share>
Successfully blocked power-save using policy: "prevent-app-suspension"
BEGIN downloading/processing video "8f16243bc1153a7b" - "Would You - Touch & Go"
Splitting video "8f16243bc1153a7b"; 4 jobs using model "htdemucs"...
Running with "-d cpu" to force CPU instead of CUDA
Trace: Error: ENOENT: no such file or directory, scandir 'C:\Users\USERNAME\AppData\Local\Temp\StemRoller-XXXXXX7wLdE0\separated\htdemucs'
    at async readdir (node:internal/fs/promises:948:18)
    at async findDemucsOutputDir (C:\share\stemroller-2.1.1-win-cuda\resources\app.asar\main-src\processQueue.cjs:188:19)
    at async _processVideo (C:\share\stemroller-2.1.1-win-cuda\resources\app.asar\main-src\processQueue.cjs:324:26)
    at async processVideo (C:\share\stemroller-2.1.1-win-cuda\resources\app.asar\main-src\processQueue.cjs:440:5) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'scandir',
  path: 'C:\\Users\\USERNAME\\AppData\\Local\\Temp\\StemRoller-XXXXXX7wLdE0\\separated\\htdemucs'
}
    at processVideo (C:\share\stemroller-2.1.1-win-cuda\resources\app.asar\main-src\processQueue.cjs:442:13)
Successfully unblocked power-save
```

```
#
# Input as online search - Ver 2.1.1 on Win11 Pro 23H2, clean install, shell as administrator
#

C:\share>.\stemroller-2.1.1-win-cuda\StemRoller.exe

C:\share>
Successfully blocked power-save using policy: "prevent-app-suspension"
BEGIN downloading/processing video "0LwcvjNJTuM" - "Lynyrd Skynyrd - Free Bird (Official Audio)"
Downloading YouTube video "0LwcvjNJTuM"; storing in "C:\Users\USERNAME\AppData\Local\Temp\StemRoller-XXXXXXeLUZiB\yt-audio"
WARNING: @distube/ytdl-core is out of date! Update with "npm install @distube/ytdl-core@latest".
Splitting video "0LwcvjNJTuM"; 4 jobs using model "htdemucs"...
Running with "-d cpu" to force CPU instead of CUDA
Trace: Error: ENOENT: no such file or directory, scandir 'C:\Users\USERNAME\AppData\Local\Temp\StemRoller-XXXXXXeLUZiB\separated\htdemucs'
    at async readdir (node:internal/fs/promises:948:18)
    at async findDemucsOutputDir (C:\share\stemroller-2.1.1-win-cuda\resources\app.asar\main-src\processQueue.cjs:188:19)
    at async _processVideo (C:\share\stemroller-2.1.1-win-cuda\resources\app.asar\main-src\processQueue.cjs:324:26)
    at async processVideo (C:\share\stemroller-2.1.1-win-cuda\resources\app.asar\main-src\processQueue.cjs:440:5) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'scandir',
  path: 'C:\\Users\\USERNAME\\AppData\\Local\\Temp\\StemRoller-XXXXXXeLUZiB\\separated\\htdemucs'
}
    at processVideo (C:\share\stemroller-2.1.1-win-cuda\resources\app.asar\main-src\processQueue.cjs:442:13)
Successfully unblocked power-save
```

**Comment by iffyloop** [comment]
Hey @lunndal, thanks for testing this so thoroughly, that's really weird that the new version is failing on both local/download, while the old one only failed with downloads. Are you by any chance able to get this up and running from source, following the instructions in the Readme? You'll need to install Node.js first, and then all other dependencies should be pulled in automatically when you run the `download-third-party-apps` script. I'm wondering if perhaps you could do some debugging on your machine, and see if there's anything at all getting written into that temporary directory before it fails. Otherwise I can try to build a different version with more debug output and send it to you. Or, if you have time and are comfortable with this, we could try to screenshare call and debug it together on your device (you can join the StemRoller Discord server and DM me to let me know you're the same user from this issue chat).

**Comment by lunndal** [comment]
@iffyloop , I'd be happy to do either one of those. How do you feel about using TeamViewer to connect to the host? It's ready for you, in a DMZ, with an admin account set up for you. Chocolatey, Chrome and Git installed. If that sounds good, just give me somewhere I can give you the credentials in a safe manner. Signal or some such?

If you'd prefer, we could go with either one of the options you suggested :)

**Comment by iffyloop** [comment]
@lunndal Thank you so much, sounds good to me! Can you DM me on Discord (I'm the admin of the StemRoller server - https://www.stemroller.com/chat), that way we can discuss privately how to exchange credentials? Signal is good, but would rather send you my contact info for it directly than post publicly here 😄

**Comment by iffyloop** [comment]
Looks like this was due to missing MSVC redistributables - should be fixed in the latest update. @ea2399  feel free to open a separate issue if you're still having the problem you described earlier after re-downloading the latest 2.1.1 and installing MSVC redistributables (if prompted).
